File: trunk/Utils/BankModel/src/main/scala/com/ligadata/Bank/SimpleModel.scala
package com.ligadata.Bank

import com.ligadata.FatafatBase.{ModelBase,ModelBaseObj,MessageContainerBase,EnvContext}
//import com.ligadata.messagescontainers._

/**
 * 
 */
object SimpleModel extends ModelBaseObj {
  
   val validMessages = Array("System.BankCustomerCodes")
   def IsValidMessage(msg: MessageContainerBase): Boolean = {
     validMessages.filter( m => m.toLowerCase == msg.FullName.toLowerCase).size > 0
   }

   def CreateNewModel(tempTransId: Long, gCtx: EnvContext, msg: MessageContainerBase, tenantId: String): ModelBase = {
     new SimpleModel(gCtx, msg, getModelName, getVersion, tenantId, tempTransId)
   }   
   def getModelName: String = { "System.SimpleModel"}  
   override def getVersion: String = {"1000011"}
   
   
   def main(args : Array[String]): Unit = {   
     var msg: System_BankCustomerCodes_1000001_1431110308663 = new System_BankCustomerCodes_1000001_1431110308663
     msg.feetype = "criminal"
     msg.feeammount = 2
     
     var myModel: ModelBase = SimpleModel.CreateNewModel(50,null, msg,"test,")
     myModel.execute(true)
   }
}

/**
 * 
 */
class SimpleModel (val gCtx : com.ligadata.FatafatBase.EnvContext, val basemsg : MessageContainerBase,
                   val modelName:String, val modelVersion:String, val tenantId: String, val tempTransId: Long) extends ModelBase {
    override def getModelName : String = SimpleModel.getModelName 
    override def getVersion : String = SimpleModel.getVersion
    override def getTenantId : String = tenantId
    override def getTempTransId: Long = tempTransId
    private val msg = basemsg.asInstanceOf[System_BankCustomerCodes_1000001_1431110308663]
    
    override def execute(emitAllResults : Boolean) : com.ligadata.FatafatBase.ModelResult = {
      doRule
      prepareResults(emitAllResults)
      null
    }  
    private def doRule: Unit = {
      println(" Executing Simple Model Rule for message " + msg.feetype + " is "+ msg.feeammount)
    }  
    private def prepareResults(emitAll: Boolean): Unit = {} 
}

File: trunk/Utils/BankModel/src/main/scala/com/ligadata/Bank/SimpleModel.java
package com.ligadata.Bank;
import com.ligadata.FatafatBase.*;


/**
 * 
 * @author dan
 *
 */
public class SimpleModel implements ModelBase {

	private MessageContainerBase msg;
	private com.ligadata.FatafatBase.EnvContext gCtx;
	private String modelName;
	private String modelVersion;
	private String tenantId;
	private long tempTransId;

	public SimpleModel (com.ligadata.FatafatBase.EnvContext pgCtx,
			MessageContainerBase basemsg, 
            String pmName,
            String pmVersion, 
            String pTenantId,
            Long pTempTransId) {
        msg = basemsg;
        gCtx = pgCtx;
        modelName = pmName;
        modelVersion = pmVersion;
        tenantId = pTenantId;
        tempTransId = pTempTransId;	
	}
	

	@Override
	@LigadataModelAnnotations(
			dependencies={"shit1","shit2"},
			modelFullName="System.SimpleModel",
			emitResults="true"
	)
	public ModelResult execute(boolean emitResults) {	
		/**************************************************/
		/**************************************************/
		/**************************************************/
		/************Rules logic goes here*****************/
		/**************************************************/
		/**************************************************/
		/**************************************************/
	
		return null;
	}

	@Override
	public String getModelName() {
		return modelName;	// Model Name
	}
	
	@Override
	public String getVersion() {
		return modelVersion;	// Model Name
	}

	@Override
	public String getTenantId() {
		return tenantId;	// Model Name
	}

	@Override
	public long getTempTransId() {
		return tempTransId;	// Model Name
	}

	@Override
	public EnvContext gCtx() {
		return gCtx;
	}

	@Override
	public MessageContainerBase msg() {
		return msg;
	}

	@Override
	public String modelName() {
		return modelName;
	}

	@Override
	public String modelVersion() {
		return modelVersion;
	}

	@Override
	public String tenantId() {
		return tenantId;
	}

	@Override
	public long tempTransId() {
		return tempTransId;
	}
	
	public static class SimpleModelObj implements ModelBaseObj {

		private static String[] validMessages = {"System.BankCustomerCodes"}; 
		private static String mName = "System.SimpleModel";
		private static String version = "1000011";
		/**
		 * IsValidMessage - required static method... if this method returns a true, then the Fatafat
		 *                  engine will create an instance of this ModelClass and execute it with the 
		 *                  message that is the parameter here.
		 * @param msg - Message received by the engine from an Input Adapter
		 * @return boolean - TRUE if msg applies to this model.
		 */
		@Override
		public boolean IsValidMessage(MessageContainerBase msg) {
			for(int i = 0; i < validMessages.length; i++) {
			    if (msg.FullName().compareToIgnoreCase(validMessages[i]) == 0)
			    	return true;
			}
			return false;
		}

		@Override
		public ModelBase CreateNewModel (long tempTransId,
				                                com.ligadata.FatafatBase.EnvContext gCtx,
				                                MessageContainerBase msg,
				                                String tenantId) {
			 return new SimpleModel(gCtx, msg, getModelName(), getVersion(), tenantId, tempTransId);
		}


		@Override
		public String getModelName() {
			return mName;
		}

		@Override
		public String getVersion() {
			return version;
		}
	}
}
